crates/db_schema/src/utils.rs

```diff
@@ -345,10 +352,32 @@ pub fn diesel_url_create(opt: Option<&str>) -> LemmyResult<Option<DbUrl>> {
   }
 }
 
+/// Sets a few additional config options necessary for starting lemmy
+fn build_config_options_uri_segment(config: &str) -> String {
+  // Set `lemmy.protocol_and_hostname` so triggers can use it
+  let lemmy_protocol_and_hostname_option =
+    "lemmy.protocol_and_hostname=".to_owned() + &SETTINGS.get_protocol_and_hostname();
+  let mut options = CONNECTION_OPTIONS.to_vec();
+  options.push(&lemmy_protocol_and_hostname_option);
+
+  let path_character = if config.contains('?') { '&' } else { '?' };
+
+  // Create the connection uri portion
+  let options_segments = &options
+    .iter()
+    .map(|o| "-c ".to_owned() + o)
+    .collect::<Vec<String>>()
+    .join(" ");
+
+  format!("{config}{path_character}options={options_segments}")
```

Comment:
The previous method was inefficient, as it ran a synchronous query to update some settings, every time a connection is made. The speed test check I added found that this was the result of the slowness, context: #5149 .

> Url

I thought about that, but chose against it for two reasons:

1. The current database uri is passed as a `&str`, and needs to go back to a string, so converting back and forth to a `Url` seems like a waste. Remember that this is also run on every single connection, so we need to stay as efficient as possible.
2. It adds a Url result / error which isn't really necessary.

Reply:
You can put it in once_cell so it only needs to be converted once.

Reply:
mmmk.

Reply:
I tried this for a while, but failed. Url::parse(any_postgres_connection) just returns `Couldn't parse postgres connection string: EmptyHost` 

I'll still try to use the once_lock so that it doesn't need to build it every time tho.

Reply:
What kind of url are you using? Seems to work fine for me.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=6e91c935eebb7e46e13fe721bdf09167

Reply:
Figured it out finally. The tests that create a dev db had illegal path characters that were unparsable.